src/Transactions/TransactionDetails/TransactionInformation/SourceValue/useSourceValue.js

```diff
@@ -0,0 +1,34 @@
+import { useQuery } from 'react-query';
+
+import { useOkapiKy } from '@folio/stripes/core';
+import {
+  LINES_API,
+  INVOICES_API,
+} from '@folio/stripes-acq-components';
+
+import { TRANSACTION_SOURCE } from '../../../constants';
+
+const queryFnsMap = {
+  [TRANSACTION_SOURCE.invoice]: async (ky, transaction) => {
+    const { folioInvoiceNo } = await ky(`${INVOICES_API}/${transaction.sourceInvoiceId}`).json();
+
+    return folioInvoiceNo;
+  },
+  [TRANSACTION_SOURCE.poLine]: async (ky, transaction) => {
+    const { poLineNumber } = await ky(`${LINES_API}/${transaction.encumbrance?.sourcePoLineId}`).json();
+
+    return poLineNumber;
+  },
+  [TRANSACTION_SOURCE.user]: () => TRANSACTION_SOURCE.user,
```

Comment:
IMHO, there is no reason to write as a query function the thing, that is not a query function. Maybe it's better to check if queryFn is defined and make query or return raw source value

Reply:
If you try to add any condition to the hook it fires a warning `React Hook "useQuery" is called conditionally. React Hooks must be called in the exact same order in every component render`

Reply:
OK, don't cal it conditionally. But the main concern is still there